src/main/java/com/shaft/api/RestActions.java

```diff
@@ -1118,6 +1119,64 @@ String prepareReportMessage(Response response, int targetStatusCode, RequestType
         return "";
     }
 
+    /**
+     * Perform Graphql Request using the "Query or Mutation" only
+     *
+     * @param base_URI The Base URI without "graphql". example: "https://api.example.com/"
+     * @param query
+     * @return Graphql Response
+     */
+    @SuppressWarnings("unchecked")
+    public static Response sendGraphqlRequest(String base_URI, String query) {
+
+        org.json.simple.JSONObject requestBody = new org.json.simple.JSONObject();
+        requestBody.put("query", query);
+
+        return buildNewRequest(base_URI, GRAPHQL_END_POINT, RestActions.RequestType.POST).setRequestBody(requestBody)
+                .setContentType(ContentType.JSON).performRequest();
+    }
+    /**
+     * Perform Graphql Request using the "Query or Mutation" and the Variables
+     *
+     * @param base_URI The Base URI without "graphql". example: "https://api.example.com/"
+     * @param query
+     * @param variables
+     * @return Graphql Response
+     */
+    @SuppressWarnings("unchecked")
+    public static Response sendGraphqlRequest(String base_URI, String query, String variables) {
+
+        org.json.simple.JSONObject requestBody = new org.json.simple.JSONObject();
+        requestBody.put("query", query);
+        requestBody.put("variables", variables);
+
+        return buildNewRequest(base_URI, GRAPHQL_END_POINT, RestActions.RequestType.POST).setRequestBody(requestBody)
+                .setContentType(ContentType.JSON).performRequest();
+    }
+
+    /**
+     * Perform Graphql Request using the "Query or Mutation", Variables, and fragments
+     *
+     * @param base_URI The Base URI without "graphql". example: "https://api.example.com/"
+     * @param query
+     * @param variables
+     * @param fragment
+     * @return Graphql Response
+     */
+    @SuppressWarnings("unchecked")
+    public static Response sendGraphqlRequest(String base_URI, String query, String variables, String fragment) {
+
+        org.json.simple.JSONObject requestBody = new org.json.simple.JSONObject();
+        requestBody.put("query", query);
+        requestBody.put("variables", variables);
+        requestBody.put("fragment", fragment);
+
```

Comment:
add a custom log message using ReportManagerHelper.logDiscrete and mention that a GraphQL request is being performed with the following parameters. This will help debug the parsing of the parameters. and it's better to use log discrete in this case because a regular log message will already be created by the performRequest method below.